src/algorand_hash.rs

```diff
@@ -43,6 +44,15 @@ impl AlgorandHash {
         }
     }
 
+    /// ## Check if all hash elements are 0
+    ///
+    /// This will return true in case all 32 bytes are 0, false otherwise.
+    /// Base64 representation of the all-zero-elements hash is
+    /// AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA=
+    pub fn is_zero(self) -> bool {
+        !self.0.iter().any(|el| *el != 0)
```

Comment:
No need to iterate over anything, just do:

```suggestion
        self == Self::default()
```